doc/reference/plutus/reference-script-example.md

```diff
@@ -0,0 +1,146 @@
+cabal install cardano-node
+./scripts/babbage/mkfiles.sh
+```
+
+To start your babbage cluster you need to run the `example/run/all.sh` shell script.
+The remainder of this guide briefly talks about the [shell script example](scripts/plutus/example-reference-script-usage.sh) that automatically does everything for you.
+
+#### Creating a reference script at a transaction output and
+#### sending ADA to the script address (with a datum)
+
+In order to use a reference script we must first create said reference script at a particular address. Note you cannot create a reference script at the same correspnding script address, this will result in a failure when trying to spend the transaction input at the script address.
+
+```bash
+cardano-cli transaction build \
+  --babbage-era \
+  --cardano-mode \
+  --testnet-magic 42 \
+  --change-address "$utxoaddr" \
+  --tx-in "$txin" \
+  --tx-out "$utxoaddr+$lovelace" \
+  --tx-out "$plutusscriptaddr+$lovelace" \
+  --tx-out-datum-hash "$scriptdatumhash" \
+  --tx-out "$dummyaddress+$lovelaceattxindiv3" \
+  --reference-script-file "$plutusscriptinuse" \
+  --protocol-params-file "$WORK/pparams.json" \
+  --out-file "$WORK/create-datum-output.body"
+```
+
+There are a couple things happening in the following `build` command.
+Firstly we are sending ADA to the plutus script address along with a datum. This is reflected in the following lines:
+
+```bash
+...
+--tx-out "$plutusscriptaddr+$lovelace" \
+--tx-out-datum-hash "$scriptdatumhash" \
+...
+```
+
+We have seen this before in the [plutus-spending-script-example.md](doc/reference/plutus/plutus-spending-script-example.md).
+
+Secondly we are creating a reference script at an address, which cannot be the same as the reference script address, of our choosing in the following lines:
+
+```bash
+...
+--tx-out "$dummyaddress+$lovelaceattxindiv3" \
+--reference-script-file "$plutusscriptinuse" \
+...
+```
+
+Specifying the `--reference-script-file` after the `--tx-out` option will construct a transaction that creates a reference script at that transaction output.
+
+We sign and then submit as usual:
+
+```bash
+cardano-cli transaction sign \
+  --tx-body-file $WORK/create-datum-output.body \
+  --testnet-magic "$TESTNET_MAGIC" \
+  --signing-key-file $UTXO_SKEY \
+  --out-file $WORK/create-datum-output.tx
+```
+
+
+#### Spending ADA at the script address
+
+Now that there is ADA at our script address, we must construct the appropriate transaction in order to spend it.
+Because we are using the `build` command we only have to concern ourselves with the following:
+
+`$plutusutxotxin` - This is the tx input that sits at the Plutus script address (NB: It has a datum hash).
+`tx-in-reference` -This specifies the reference input you are using to witness a transaction input.
+`plutus-script-v2`- This specifies the version of the reference script at the reference input. There are no restrictions on the type of script we can have at a reference input.
```

Comment:
Right, but it's always invalid to execute a referenced V1 script. We should make this limitation clear.

Reply:
When using `--plutus-script-v1` I always see `Command failed: transaction build  Error: Error translating the transaction context: ReferenceInputsNotSupported`, so should it be an option?

Reply:
Right so there is some confusion here. It was my understanding that we can use V1 scripts as reference scripts to witness the spending of a particular tx input. Is this the case @JaredCorduan ?

Reply:
If it is, we shouldn't get `ReferenceInputsNotSupported` when using a V1 reference script to witness a tx input

Reply:
> 



> Right so there is some confusion here. It was my understanding that we can use V1 scripts as reference scripts to witness the spending of a particular tx input. Is this the case @JaredCorduan ?

it is not the case, the context translation will fail.